lib/hooks.js

```diff
@@ -114,46 +114,43 @@ const Hooks = {
   },
 
   runHooks(hooks) {
-    const fnArgs = Utils.sliceArgs(arguments, 1);
+    if (!hooks) throw new Error('runHooks requires atleast 1 argument');
+
+    const hookArgs = Utils.sliceArgs(arguments, 1);
     let hookType;
 
     if (typeof hooks === 'string') {
       hookType = hooks;
       hooks = this.options.hooks[hookType] || [];
-      if (!Array.isArray(hooks)) hooks = hooks === undefined ? [] : [hooks];
       if (this.sequelize) hooks = hooks.concat(this.sequelize.options.hooks[hookType] || []);
     }
 
     if (!Array.isArray(hooks)) {
-      hooks = hooks === undefined ? [] : [hooks];
+      hooks = [hooks];
     }
 
-    // run hooks as sync functions if flagged as sync
+    // synchronous hooks
     if (hookTypes[hookType] && hookTypes[hookType].sync) {
```

Comment:
Is this really needed? If the hook is sync, then `Promise.resolve()` will resolve immediately